### PR TITLE
Fix issue with non-terminating gcd/lcm

### PR DESCRIPTION
**Pull request by jfalcou**
Issues on ARM v7 comes from cases where we miss a bit in the result of rem and the Euler algorithm can't converge.

Fixing rem allows for proper completion.

### CODE REVIEW

File: include/eve/function/gcd.hpp [hunk 1]
```diff
@@ -43,20 +43,11 @@ namespace eve
   //!
   //!@warning
   //!    `p` and `n` can be of any [real values](@ref eve::value) type, but when the types are not integral
-  //!    the greatest common divisor is defined only if `p` and `n` element are [flint](@ref eve::is_flint). If it is not the
-  //!    case the corresponding result will be Nan.
+  //!    the greatest common divisor is defined only if `p` and `n` element are [flint](@ref eve::is_flint). If any is not flint
+  //!    the function will assert.
```
Comment (jfalcou) [inline include/eve/function/gcd.hpp:47]: assert is not a specification. Prefers:
"If any of the arguments is flint, then the behavior is undefined."

File: include/eve/function/lcm.hpp [hunk 2]
```diff
@@ -43,8 +43,8 @@ namespace eve
   //!
   //!@warning
   //!    `p` and `n` can be of any [real values](@ref eve::value) type, but when the types are not integral
-  //!    the least common multiple is defined only if `p` and `n` element are [flint](@ref eve::is_flint). If it is not the
-  //!    case the corresponding result will be Nan.
+  //!    the least common multiple is defined only if `p` and `n` element are [flint](@ref eve::is_flint). If any
+  //!    is not flint the result is UB.
```
Comment (jfalcou) [inline include/eve/function/lcm.hpp:47]: We don't pay the doc per character: "If any of the arguments is flint, then the behavior is undefined."

File: include/eve/module/real/combinatorial/function/regular/generic/gcd.hpp [hunk 3]
```diff
@@ -103,29 +109,25 @@ namespace eve::detail
...
   template<floating_value T>
-  EVE_FORCEINLINE auto gcd_(EVE_SUPPORTS(cpu_), T a, T b) noexcept
+  auto gcd_(EVE_SUPPORTS(cpu_), T a, T b) noexcept
   {
     if constexpr( has_native_abi_v<T> )
     {
-      auto valid = is_flint(a) && is_flint(b);
+      a = abs(a);
+      b = abs(b);
+      EVE_ASSERT(eve::all(is_flint(a) && is_flint(b)), "gcd: some entries are not flint");
```
Comment (jfalcou) [inline include/eve/module/real/combinatorial/function/regular/generic/gcd.hpp:118]: `EVE_ASSERT` before any operations.

File: include/eve/module/real/combinatorial/function/regular/generic/lcm.hpp [hunk 4]
```diff
@@ -30,46 +32,21 @@ namespace eve::detail
...
 
   template<real_value T>
-  EVE_FORCEINLINE T lcm_(EVE_SUPPORTS(cpu_), T a, T b) noexcept
+  T lcm_(EVE_SUPPORTS(cpu_), T a, T b) noexcept
   {
     if constexpr( has_native_abi_v<T> )
     {
+      a = abs(a);
+      b = abs(b);
+      EVE_ASSERT(eve::all(is_flint(a) && is_flint(b)), "lcm: some entries are not flint");
```
Comment (jfalcou) [inline include/eve/module/real/combinatorial/function/regular/generic/lcm.hpp:41]: `EVE_ASSERT` before any operations

File: test/unit/module/real/combinatorial/gcd.cpp [hunk 5]
```diff
@@ -27,25 +27,24 @@ EVE_TEST_TYPES( "Check return types of eve::gcd"
...
-};
-
-//==================================================================================================
-// Test for corner-cases values
-//==================================================================================================
+// auto mini = []< typename T>(eve::as<T> const &){return eve::valmin(eve::as<T>())+1; };
+
+// EVE_TEST( "Check corner-cases behavior of eve::gcd on wide"
+//         , eve::test::simd::integers
+//         , eve::test::generate ( eve::test::randoms(mini, eve::valmax)
```
Comment (jfalcou) [inline test/unit/module/real/combinatorial/gcd.cpp:34]: Remove don't comment
